components/server/src/workspace/workspace-starter.ts

```diff
@@ -1370,11 +1370,11 @@ export class WorkspaceStarter {
             }
         }
 
-        let volumeSnapshotId = lastValidWorkspaceInstanceId;
-        // if this is snapshot or prebuild context, then try to find volume snapshot id in it
-        if (SnapshotContext.is(workspace.context) || WithPrebuild.is(workspace.context)) {
-            volumeSnapshotId = workspace.context.snapshotBucketId;
-        }
+        const volumeSnapshotId =
+            SnapshotContext.is(workspace.context) || WithPrebuild.is(workspace.context)
+                ? workspace.context.snapshotBucketId
+                : lastValidWorkspaceInstanceId;
+
         let volumeSnapshotInfo = new VolumeSnapshotInfo();
         const volumeSnapshots = await this.workspaceDb.trace(traceCtx).findVolumeSnapshotById(volumeSnapshotId);
```

Comment:
@sagor999 the only other thing I can think of, if this doesn't work, is that maybe `findVolumeSnapshotById` is mutating the volumeSnapshotId value. :shrug: